synapse/federation/sender/per_destination_queue.py

```diff
@@ -346,6 +387,8 @@ async def _transaction_transmission_loop(self) -> None:
                 e.code,
                 e,
             )
+
+            # XXX REVIEW should we be catching up?
```

Comment:
I suppose I want to be more clear on which of these exceptions, we should later try to catch up again, or which ones we should treat as 'actually drop the PDUs for good this time'?

After looking with a fresh mind, I think yes, we should enable catch-up on: `RequestSendFailed`, `HttpResponseException`

Not sure about: `FederationDeniedError`, `Exception`

Thoughts?

Reply:
yes, I think we should enter "catchup" mode in most of these cases.

We should *not* do so for `FederationDeniedError` since that means the remote server is deliberately blocked in our whitelist.

All the other cases should trigger a catchup afaict.